src/components/mainContent/PostsList.js

```diff
@@ -1,22 +1,82 @@
-import React from "react";
+import React, { useState } from "react";
 import Post from "../posts/Post";
 import CommentSection from "../comments/CommentSection";
-
+import ForwardButton from "./ForwardButton";
+import "./mainContent.css";
 export default function PostsList(props) {
-  /*This function takes the list of post/followups from our backend and renders it accordingly
-Sections can either be posts or followups (Interactive sections might be seperate in future as well*/
+  const [visibleSections, setVisibleSections] = useState(0);
+  const [currentVisibleText, setCurrentVisibleText] = useState(0);
+  const [blockRendering, setBlockRendering] = useState(false);
+  const [forwardVisible, setForwardVisible] = useState(false);
+
+  /*This function takes the list of post/followups, takes all the previousSections that are already loaded,
+  and makes them show all their text*/
+  let completedContent = props.content
+    .slice(0, visibleSections)
+    .map((section, index) => {
+      /*Not using ternaries since there might be 3 possibilities if we make
+      interactive sections seperate from posts in the future
+      */
+      if (section.post)
+        return (
+          <Post
+            key={index}
+            postContent={section.post}
+            currentVisibleText={section.post.bodyText.length}
+          />
+        );
+      if (section.followups)
+        return <CommentSection key={index} followups={section.followups} />;
+      return "";
+    });
+  /* This function takes the section that the user is currently on, and makes it so that its content
+    can change depending on how the DirectionButtons are pressed*/
+  let currentContent = props.content
```

Comment:
This is somewhat duplicated with `completedContent`, though I see that there's a different use of `blockRendering` - still, I think there are different workarounds that you can use to avoid duplicating too much code. We can revisit this later if you'd like.

Reply:
I kept `completedContent` and `currentContent` separate so that only `currentContent` gets affected by changes in `visibleText` in state. `completedContent` will have all its content visible, while currentContent changes with `handleForward` in `ForwardButton`